sw/tools/generators/gen_airframe.ml

```diff
@@ -337,9 +337,17 @@ let parse_ap_only_commands = fun out ap_only ->
 
 let parse_command = fun out command no ->
   let command_name = "COMMAND_"^ExtXml.attrib command "name" in
-  define_out out command_name (string_of_int no);
   let failsafe_value = int_of_string (ExtXml.attrib command "failsafe_value") in
-  { failsafe_value = failsafe_value; foo = 0}
+  let actuator_type = ExtXml.attrib_or_default command "actuator_type" "REAL" in 
+  define_out out command_name (string_of_int no);
+  { failsafe_value = failsafe_value; foo = 0 ; actuator_type = actuator_type }  
+
+let count_commands_by_type commands_params =
+  List.fold_left (fun acc cmd ->
```

Comment:
You could use `Array.fold_left` to avoid an extra conversion between list and array type

Reply:
Thanks, done that!